src/core/lib/surface/completion_queue.cc

```diff
@@ -48,6 +49,9 @@ grpc_tracer_flag grpc_trace_cq_refcount =
     GRPC_TRACER_INITIALIZER(false, "cq_refcount");
 #endif
 
+GPR_TLS_DECL(g_cq_first_event);
```

Comment:
Please add a comment above indicating what is the purpose of these TLS (implementation of cq thread-local cache, etc)

Reply:
Since the cache implementation here is essentially a single element in a single CQ, these variable names are misleading. Can you call it `g_cached_event` and `g_cached_cq` or something like that? I would also add a comment saying that in the current implementation, only 1 event for 1 CQ is ever cached at a thread, to make it clear that this is the current cache characteristic. This is quite fine for the use case, but the name "cache" suggests the possibility of more than that (which is fine from an API perspectiv, since the current API does not preclude that in the unlikely event that it is ever desired in the future.)